### ISSUE TITLE
[HOLD for payment 2024-04-05] [$500] Chat - 3 dot menu overlaps with the emoji menu

### ISSUE THREAD

**Issue by izarutskaya** [Expensify/App#37266]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 1.4.44-0
**Reproducible in staging?:** Y
**Reproducible in production?:** Y
**Email or phone of affected tester (no customers):** applausetester+emilio@applause.expensifail.com
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Issue reported by:** Applause-Internal Team

## Action Performed:
1. Access staging.new.expensify.com
2. Sign into a valid account
3. Go to any chat
4. Send any message
5. Hover over the message and tap on the 3 dot menu

## Expected Result:
User expects a modal to open but to be able to still see the previous modal

## Actual Result:
The two modals are overlapping

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android: Native
- [ ] Android: mWeb Chrome
- [ ] iOS: Native
- [ ] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [x] MacOS: Desktop

## Screenshots/Videos

![Bug6393588_1709009088248!Overlapping_when_tapping_on_3_dot_menu_](https://github.com/Expensify/App/assets/115492554/6cba54e6-d985-4ef1-ac5d-38bcc793c523)

</details>

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~017349bd168876c6ff</li>
        <li>Upwork Job ID: 1762967904272871424</li>
        <li>Last Price Increase: 2024-03-13</li>
<li>Automatic offers: </li>
<ul>
<li>DylanDylann | Reviewer | 0</li>
<li>dukenv0307 | Contributor | 0</li>
</ul></li>
    </ul>
</details>

**Comment by izarutskaya** [comment]
@jliexpensify I haven't added the External label as I wasn't 100% sure about this issue. Please take a look and add the label if you agree it's a bug and can be handled by external contributors.

**Comment by izarutskaya** [comment]
We think that this bug might be related to #vip-vsb
CC @quinthar

**Comment by jliexpensify** [comment]
Yep, can repro and I think this should go in #vip-vsb

**Comment by dubielzyk-expensify** [comment]
Hold off on fixing this until we get final confirmation. 

My comment was that I don't think this is a bug because you can still hover other comments below and see the comment action bar. So unless we wanted to disable that, the overlay should stay as is.

Though if we wanted to we could lower the overlay in the Y position to make sure it covers the comment action menu of the comment you've clicked on. I'm checking with @Expensify/design to see if they have strong feelings.

**Comment by jliexpensify** [comment]
Sweet, thanks @dubielzyk-expensify - have dropped a `hold` on the title!

**Comment by shawnborton** [comment]
This is what I wrote in Slack: Oh interesting... I would expect this context menu to launch below the overflow icon if there is space, and if not, above the overflow icon in a way that wouldn't obscure the menu behind it at all. So I personally do think this is a bug.

**Comment by dannymcclain** [comment]
I agree with Shawn. I would expect it to sit in the same position as the emoji picker (but on the bottom in there's room).

Weren't we just talking about this in another issue a week or so ago?

![CleanShot 2024-02-28 at 08 14 29@2x](https://github.com/Expensify/App/assets/7516624/7e680978-b433-49d5-8f96-3b3c9ea64ea8)

Might be worth showing that like the emoji menu, this menu should have automatic viewport collision detection (which it seems like it already does).

![CleanShot 2024-02-28 at 08 15 22@2x](https://github.com/Expensify/App/assets/7516624/01f6ab4f-204f-4e60-a344-b1bbc1fa1f7f)

**Comment by jliexpensify** [comment]
Determined this is a bug. Input from @shawnborton:

>I would expect this context menu to launch below the overflow icon if there is space, and if not, above the overflow icon in a way that wouldn't obscure the menu behind it at all

**Comment by dubielzyk-expensify** [comment]
Wonderful work @usman-ghani564, that looks great to me. Just to confirm that the spacing between the overflow menu is `8px`? 

Keen to get @Expensify/design feedback as well before settling in on this.

**Comment by dubielzyk-expensify** [comment]
Also I do see that the overflow menu is completely flush with the edge of the screen on the right hand side. Could we make sure that it's at least 20px from the edge? Looks a bit tight in the screenshot:
![CleanShot 2024-02-29 at 13 52 29@2x](https://github.com/Expensify/App/assets/145416827/ffaabf35-62ab-4034-b8ae-0dd546e4b507)

**Comment by shawnborton** [comment]
Interesting, I would think this one should always launch respective of the overflow icon and not the mouse position. And it should be aligned on the right edge of the mini menu.

**Comment by dannymcclain** [comment]
>Interesting, I would think this one should always launch respective of the overflow icon and not the mouse position. And it should be aligned on the right edge of the mini menu.

+1.

Is there a way for us to make all our popovers (that are triggered from a button, not a right-click) have consistent positions relative to the button that spawned them? Looking at the screenshots on this issue, the position/spacing looks different from the Emoji menu which also looks different from the three-dot overflow at the top. **What's the rule?** Always `8px` away from the trigger?

![CleanShot 2024-02-29 at 08 15 42@2x](https://github.com/Expensify/App/assets/7516624/d69869a7-269b-470a-93db-421feeb7179c)

![CleanShot 2024-02-29 at 08 15 30@2x](https://github.com/Expensify/App/assets/7516624/e47866bb-7a68-4a93-bdb5-556838d0bec3)

**Comment by dubielzyk-expensify** [comment]
Yeah, +1 to what @dannymcclain and @shawnborton said. The screenshot above looks too far off now. Basically where the red outline is below is where we want it to line up:
![CleanShot 2024-03-01 at 08 57 59@2x](https://github.com/Expensify/App/assets/145416827/4af6b6dc-94b5-49ea-8b72-051549d74a93)

But like Danny said, this should be consistent with other parts of the app. It should be tied to the 3 dots and where the icon is, not the click itself.

Does that make sense?

**Comment by DylanDylann** [comment]
@dubielzyk-expensify I think the horizontal position of the context menu is aligned with the position of a three-dot menu in the header

https://github.com/Expensify/App/assets/141406735/c1c85349-4c52-47bc-95c8-f9802ecea4dd

**Comment by DylanDylann** [comment]
@Expensify/design Let's see this case: Click three-dot to open the three-dot menu and move the cursor to the above message. See that the three-dot menu overlaps the context menu action. What do you think about the expected for this case?

https://github.com/Expensify/App/assets/141406735/103cb65e-ce0a-44b4-ae68-62dae7ff17e0

**Comment by dannymcclain** [comment]
>Click three-dot to open the three-dot menu and move the cursor to the above message. See that the three-dot menu overlaps the context menu action. What do you think about the expected for this case?

IMO this is expected/desired behavior for a popover menu. Click to open, click to close.

**Comment by dubielzyk-expensify** [comment]
> @dubielzyk-expensify I think the horizontal position of the context menu is aligned with the position of a three-dot menu in the header

Looks good.

> Click three-dot to open the three-dot menu and move the cursor to the above message. See that the three-dot menu overlaps the context menu action. What do you think about the expected for this case?

Agree with Danny that this is expected behavior

**Comment by DylanDylann** [comment]
@dukenv0307 @usman-ghani564 Please update the proposal to include the fix for emoji popover

@dukenv0307 One more thing, please help to detail your implementation

**Comment by DylanDylann** [comment]
I prefer @dukenv0307's [idea](https://github.com/Expensify/App/issues/37266#issuecomment-1970352130) but waiting him for detailing his solution

**Comment by DylanDylann** [comment]
@usman-ghani564 With your solution, the position of the popover will rely on the position of the cursor when clicking the three-dot button. I think the position of the popover only should rely on three-dot button component as the anchor

**Comment by DylanDylann** [comment]
I still evaluate proposals

**Comment by DylanDylann** [comment]
Both proposals could resolve this issue. 
@usman-ghani564 Thanks for your effort. But from the UX perspective, I prefer @dukenv0307's [proposal](https://github.com/Expensify/App/issues/37266#issuecomment-1970352130) with this suggestion: the popover's position should only rely on the three-dot button component as the anchor. And his implementation also looks better

🎀 👀 🎀 C+ reviewed

**Comment by jliexpensify** [comment]
Not overdue, chill Melvin!

**Comment by jliexpensify** [comment]
Hello @sonialiap - I am headed OOO from the 21st to 31st March so have used the auto-assigner for a buddy.

Assuming all goes well:

PAYMENT SUMMARY:

- C: @dukenv0307: $500
- C+: @DylanDylann $500

[Upwork job](https://www.upwork.com/jobs/~017349bd168876c6ff)

- PR is being worked on atm
- Payment may need to be made before I get back

Thanks!

**Comment by jliexpensify** [comment]
Back, unassigning Sonia!

**Comment by jliexpensify** [comment]
PAYMENT SUMMARY:

C: @dukenv0307: $500
C+: @DylanDylann $500 - bumping for checklist!

[Upwork job](https://www.upwork.com/jobs/~017349bd168876c6ff)

**Comment by DylanDylann** [comment]
BugZero Checklist: The PR fixing this issue has been merged! The following checklist ([instructions](https://stackoverflow.com/c/expensify/questions/14574)) will need to be completed before the issue can be closed:

 [@DylanDylann] The PR that introduced the bug has been identified. Link to the PR: **NA**
 [@DylanDylann] The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake. Link to comment: **NA**
 [@DylanDylann] A discussion in #expensify-bugs has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner. Link to discussion: **NA**
 [@DylanDylann] Determine if we should create a regression test for this bug. **Yes**
 [@DylanDylann] If we decide to create a [regression test](https://github.com/Expensify/App/blob/main/contributingGuides/REGRESSION_TEST_BEST_PRACTICES.md) for the bug, please propose the regression test steps to ensure the same bug will not reach production again.


**Regression Test Proposal**
1. Open a report
2. Send a message
3. Hover over the message and tap on three-dot menu
4. Verify that the menu is displayed above or below the three-menu icon and it doesn't overlap the mini context menu


Do we agree 👍 or 👎

**Comment by jliexpensify** [comment]
Paid and job closed!
